cmd/opentelemetry-collector/app/defaults/default_config.go

```diff
@@ -0,0 +1,109 @@
+	"github.com/open-telemetry/opentelemetry-collector/receiver"
+	"github.com/open-telemetry/opentelemetry-collector/receiver/jaegerreceiver"
+
+	"github.com/jaegertracing/jaeger/cmd/opentelemetry-collector/app/exporter/cassandra"
+	"github.com/jaegertracing/jaeger/cmd/opentelemetry-collector/app/exporter/elasticsearch"
+	"github.com/jaegertracing/jaeger/cmd/opentelemetry-collector/app/exporter/kafka"
+)
+
+// DefaultConfig creates default configuration.
+// It enabled default Jaeger receivers, processors and exporters.
+func DefaultConfig(storageType string, factories config.Factories) *configmodels.Config {
+	exporters := createExporters(storageType, factories)
+	types := []string{}
+	for _, v := range exporters {
+		types = append(types, v.Type())
+	}
+	return &configmodels.Config{
+		Receivers:  createReceivers(factories),
+		Exporters:  exporters,
+		Processors: createProcessors(factories),
+		Service: configmodels.Service{
+			Pipelines: map[string]*configmodels.Pipeline{
+				"traces": {
+					InputType:  configmodels.TracesDataType,
+					Receivers:  []string{"jaeger"},
+					Exporters:  types,
+					Processors: []string{"batch"},
+				},
+			},
+		},
+	}
+}
+
+func createReceivers(factories config.Factories) configmodels.Receivers {
+	rec := factories.Receivers["jaeger"].CreateDefaultConfig().(*jaegerreceiver.Config)
+	// TODO load and serve sampling strategies
+	// TODO bind sampling strategies file
+	rec.Protocols = map[string]*receiver.SecureReceiverSettings{
+		"grpc": {
+			ReceiverSettings: configmodels.ReceiverSettings{
+				Endpoint: "localhost:14250",
+			},
+		},
+		"thrift_http": {
+			ReceiverSettings: configmodels.ReceiverSettings{
+				Endpoint: "localhost:14268",
+			},
+		},
+		"thrift_compact": {
+			ReceiverSettings: configmodels.ReceiverSettings{
+				Endpoint: "localhost:6831",
+			},
+		},
+		"thrift_binary": {
+			ReceiverSettings: configmodels.ReceiverSettings{
+				Endpoint: "localhost:6832",
+			},
+		},
+	}
+	return map[string]configmodels.Receiver{
+		"jaeger": rec,
+	}
+}
+
+func createExporters(storageTypes string, factories config.Factories) configmodels.Exporters {
+	exporters := configmodels.Exporters{}
+	for _, storage := range strings.Split(storageTypes, ",") {
+		if storage == "elasticsearch" {
+			es := factories.Exporters[elasticsearch.TypeStr].CreateDefaultConfig()
+			exporters[elasticsearch.TypeStr] = es
+		} else if storage == "cassandra" {
+			cass := factories.Exporters[cassandra.TypeStr].CreateDefaultConfig()
+			exporters[cassandra.TypeStr] = cass
+		} else if storage == "kafka" {
+			kaf := factories.Exporters[kafka.TypeStr].CreateDefaultConfig()
+			exporters[kafka.TypeStr] = kaf
+		}
+	}
+	return exporters
+}
+
+func createProcessors(factories config.Factories) configmodels.Processors {
+	batch := factories.Processors["batch"].CreateDefaultConfig().(*batchprocessor.Config)
+	return map[string]configmodels.Processor{
+		"batch": batch,
```

Comment:
Good point - the other reason for using `queued_retry` is that it will have some of the metrics we are interested in using.

Reply:
I have used the batch as a starting point since there has to be at least one processor. In the production-ready settings we might run batch, retry and memory limiter. Let's handle this in a separate task.

> Which brings about a more important question of - what are the use cases of running the otel-collector without a configuration file? In what scenario would we want to do that?

The highest priority we had was to be a drop-in replacement for the current collector. The OTEL collector fails without the config file which is not compatible behavior with our collector.

Reply:
https://github.com/jaegertracing/jaeger-opentelemetry-collector/issues/38

Reply:
> there has to be at least one processor

IIRC, this is no longer true, but yes I agree, let's start with the batch processor and merge this soon.